src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -612,22 +612,30 @@ public virtual string ToString(TextSpan span)
             CheckSubSpan(span);
 
             // default implementation constructs text using CopyTo
-            var builder = new StringBuilder();
-            var buffer = new char[Math.Min(span.Length, 1024)];
+            var builder = PooledStringBuilder.GetInstance();
+            var buffer = s_charArrayPool.Allocate();
 
-            int position = Math.Max(Math.Min(span.Start, this.Length), 0);
-            int length = Math.Min(span.End, this.Length) - position;
+            try
+            {
+                int position = Math.Max(Math.Min(span.Start, this.Length), 0);
+                int length = Math.Min(span.End, this.Length) - position;
+                builder.Builder.EnsureCapacity(length);
 
-            while (position < this.Length && length > 0)
+                while (position < this.Length && length > 0)
+                {
+                    int copyLength = Math.Min(buffer.Length, length);
+                    this.CopyTo(position, buffer, 0, copyLength);
+                    builder.Builder.Append(buffer, 0, copyLength);
+                    length -= copyLength;
+                    position += copyLength;
+                }
+            }
+            finally
```

Comment:
I'm not really sure of the need to try/finally this, but not try/finally the string builder release? Feels like it can just be removed.

Reply:
That makes sense. This codepath shouldn't be throwing, and it's not the end of the world if the pool loses track of an object if it does.